### ISSUE TITLE
Script ojs-start problems

### ISSUE THREAD

**Issue by matgn** [pkp/docker-ojs#179]
- Running cron in background is not recommended as the process is not monitored. I saw that the application is build to use scheduled tasks "via a mechanism like UNIX’s cron", I think it would be a better approach to schedule directly via the application or an other component (cronjob for k8s / one other container for docker)
https://docs.pkp.sfu.ca/ojs-2-technical-reference/en/scheduled_tasks

- The script do not execute the commands after the line : 
`/usr/sbin/httpd -f /etc/apache2/httpd.conf -DFOREGROUND`
(Container follows the apache process) 
So the tail command and cli install couldn't be launch. Anyway, the cli_install is just a curl. It can't be used during the "installation" part. The end of the installation must be the running web server.
This curl command may be launch after container created. For k8s -> with a Job + hook

**Comment by matgn** [comment]
Second point relative to #52

**Comment by marcbria** [comment]
TL;DR; it's feasible but not a priority right now.

The idea behind this project is to create simple self-contained containers.

If cron inside the main container is not playing gentle with k8s, we can move it into a different container and call it from docker-compose.yml.

But before starting, a little research over crontab projects needs to be done to select the right stack (a simple cron? [crontab-ui](https://github.com/alseambusher/crontab-ui)? [ofelia](https://github.com/mcuadros/ofelia)? ...). The solution need to take in consideration that users would like to building a farm of journals without k8s (centralized cron-ui?), but in case of doubts, I will always suggest to use of the most standard solution.

Let us some time to finish with the refactor before we go into this.

Cheers,
m.

**Comment by matgn** [comment]
> TL;DR; it's feasible but not a priority right now.
> 
> The idea behind this project is to create simple self-contained containers.
> 
> If cron inside the main container is not playing gentle with k8s, we can move it into a different container and call it from docker-compose.yml.
> 
> But before starting, a little research over crontab projects needs to be done to select the right stack (a simple cron? [crontab-ui](https://github.com/alseambusher/crontab-ui)? [ofelia](https://github.com/mcuadros/ofelia)? ...). The solution need to take in consideration that users would like to building a farm of journals without k8s (centralized cron-ui?), but in case of doubts, I will always suggest to use of the most standard solution.
> 
> Let us some time to finish with the refactor before we go into this.
> 
> Cheers, m.

What about Point 2 ? 

Is it possible to take a look at this refactor project ?

**Comment by marcbria** [comment]
Nothing solid yet.

To deal with the dockerHub auto-bulid shutdown, we moved to gitLab (to test ci/cd pipelines a little):
https://gitlab.com/pkp-org/docker/ojs/

It made us modify our building script to fit better with a real ci/cd pipeline (new images are autobuild when a new tag is commit).

At same time we are talking about what images do we need in this RFC:
https://github.com/pkp/docker-ojs/issues/159

We will probably join in a jitsi at the beginning of next month to get a more clear picture of all this.
A summary of the meeting will be published, but let me know if you like to join.

Cheers,
m.

**Comment by matgn** [comment]
> We will probably join in a jitsi at the beginning of next month to get a more clear picture of all this. A summary of the meeting will be published, but let me know if you like to join.
> 
> Cheers, m.

I'd be happy to join, yes.

**Comment by luisfelipe3d** [comment]
> We will probably join in a jitsi at the beginning of next month to get a more clear picture of all this.
> A summary of the meeting will be published, but let me know if you like to join.

I'II like to join, please, tell me when.

**Comment by matgn** [comment]
> We will probably join in a jitsi at the beginning of next month to get a more clear picture of all this. A summary of the meeting will be published, but let me know if you like to join.

Any update ?

**Comment by marcbria** [comment]
We are thinking in how it all need to be.
A document will be published soon with some ideas/proposals to know how it need to be implemented.
I will post this document in the RFC issue (#159) looking for community feedback.
This document will be the basis of discussion for a virtual meeting that I hope we will be able to arrange before the end of the month.

**Comment by marcbria** [comment]
Following here:
https://github.com/pkp/containers/issues/9
